File: commons/dto/src/main/java/net/blueshell/dto/ContributionPeriodDTO.java
package net.blueshell.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

@Data
@EqualsAndHashCode(callSuper = false)
public class ContributionPeriodDTO extends BaseDTO {
    private Long id;

    private LocalDate startDate;

    private LocalDate endDate;

    private double halfYearFee;

    private double fullYearFee;

    private double alumniFee;

    private Long listId;
}

File: api/src/main/kotlin/net/blueshell/api/feature/contribution/dto/ContributionPeriodDTO.kt
package net.blueshell.api.feature.contribution.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotNull
import net.blueshell.api.shared.dto.AuditedAutoIdDTO
import java.time.LocalDate

@Schema(name = "ContributionPeriod")
data class ContributionPeriodDTO(
    @field:NotNull
    var startDate: LocalDate,

    @field:NotNull
    var endDate: LocalDate,

    @field:NotNull
    var halfYearFee: Double = 0.0,

    @field:NotNull
    var fullYearFee: Double = 0.0,

    @field:NotNull
    var alumniFee: Double = 0.0,

    var listId: Long? = null
) : AuditedAutoIdDTO()
